// landing/src/app/compare/page.tsx
import { Shield, Search, Code2, Box, ArrowRight } from 'lucide-react'
import Link from 'next/link'

export default function CompareHubPage() {
  const comparisons = [
    {
      title: 'Spectra vs. grep',
      description: 'Why regular expressions aren\'t enough to discover cryptographic primitives.',
      href: '/compare/vs-grep',
      icon: Search,
      color: 'text-emerald-400'
    },
    {
      title: 'Spectra vs. DIY',
      description: 'Why you shouldn\'t spend 6 months building an internal AST scanner.',
      href: '/compare/vs-diy',
      icon: Code2,
      color: 'text-blue-400'
    },
    {
      title: 'Spectra vs. Generic SCA',
      description: 'How Snyk/Trivy and Spectra solve different problems entirely.',
      href: '/compare/vs-sca',
      icon: Box,
      color: 'text-purple-400'
    }
  ]

  return (
    <div className="min-h-screen bg-slate-950 text-slate-300 py-20 px-6">
      <div className="max-w-5xl mx-auto">
        <Link href="/" className="inline-flex items-center gap-2 text-indigo-400 hover:text-indigo-300 font-bold mb-12">
          &larr; Back to Spectra
        </Link>
        
        <div className="text-center mb-16">
          <Shield className="w-16 h-16 text-indigo-500 mx-auto mb-6" />
          <h1 className="text-5xl font-bold text-white mb-6">Compare Spectra</h1>
          <p className="text-xl text-slate-400 max-w-2xl mx-auto">
            Spectra is building a new category: <strong>Cryptographic Intelligence</strong>. 
            Here is how it compares to existing workflows and tools.
          </p>
        </div>

        <div className="grid md:grid-cols-3 gap-6">
          {comparisons.map((item, i) => {
            const Icon = item.icon
            return (
              <Link key={i} href={item.href} className="block group">
                <div className="h-full bg-slate-900 border border-slate-800 hover:border-indigo-500/50 p-8 rounded-2xl transition-all duration-300">
                  <div className={`w-12 h-12 rounded-xl bg-slate-800 flex items-center justify-center mb-6 \${item.color}`}>
                    <Icon className="w-6 h-6" />
                  </div>
                  <h2 className="text-xl font-bold text-white mb-3 group-hover:text-indigo-400 transition-colors">{item.title}</h2>
                  <p className="text-slate-400 mb-6">{item.description}</p>
                  <div className="flex items-center gap-2 text-indigo-400 font-medium">
                    Read comparison <ArrowRight className="w-4 h-4 group-hover:translate-x-1 transition-transform" />
                  </div>
                </div>
              </Link>
            )
          })}
        </div>
      </div>
    </div>
  )
}
