// landing/src/app/roadmap/page.tsx
import { Map, CheckCircle2, Circle, Clock } from 'lucide-react'
import Link from 'next/link'

export default function RoadmapPage() {
  const phases = [
    {
      title: 'Phase 1: The Core Engine',
      status: 'completed',
      date: 'Completed',
      items: [
        'Polyglot detection (Go, Python, Java, JS, Rust, C++)',
        'Key size and contextual curve extraction',
        'Basic QRS (Quantum Risk Score) scoring algorithm',
        'Initial CLI binary structure'
      ]
    },
    {
      title: 'Phase 2: The Intelligence Platform',
      status: 'completed',
      date: 'Completed',
      items: [
        'SQLite local persistence and scanning state',
        'Cryptographic Agility Index (CAI) calculation',
        'Network endpoint TLS scanning (`spectra endpoint`)',
        'Compliance mapping against CNSA 2.0 & PCI DSS v4.0'
      ]
    },
    {
      title: 'Phase 3: Integration & Distribution',
      status: 'in-progress',
      date: 'Current',
      items: [
        'Interactive Web Playground and WASM build',
        'VS Code Extension (spectra-security)',
        'GitHub Pull Request App (Spectra CI)',
        'CycloneDX 1.7 CBOM Visualizer & Strategy Brief tools'
      ]
    },
    {
      title: 'Phase 4: The Enterprise Platform',
      status: 'planned',
      date: 'Upcoming',
      items: [
        'Multi-tenant SSO web dashboard',
        'Historical Risk Velocity tracking & alerts',
        'SIEM integration (Splunk, Datadog)',
        'Automated Pull Request generation for trivial migrations'
      ]
    }
  ]

  return (
    <div className="min-h-screen bg-slate-950 text-slate-300 py-20 px-6">
      <div className="max-w-4xl mx-auto">
        <Link href="/" className="inline-flex items-center gap-2 text-indigo-400 hover:text-indigo-300 font-bold mb-12">
          &larr; Back to Spectra
        </Link>
        
        <div className="text-center mb-16">
          <Map className="w-16 h-16 text-indigo-500 mx-auto mb-6" />
          <h1 className="text-5xl font-bold text-white mb-6">Public Roadmap</h1>
          <p className="text-xl text-slate-400 max-w-2xl mx-auto">
            The journey from a standalone CLI to a global Cryptographic Intelligence category.
          </p>
        </div>

        <div className="space-y-8">
          {phases.map((phase, i) => (
            <div key={i} className={`bg-slate-900 border \${phase.status === 'in-progress' ? 'border-indigo-500' : 'border-slate-800'} p-8 rounded-2xl relative overflow-hidden`}>
              {phase.status === 'in-progress' && (
                <div className="absolute top-0 right-0 bg-indigo-500 text-white text-xs font-bold px-4 py-1 rounded-bl-lg">
                  CURRENT FOCUS
                </div>
              )}
              
              <div className="flex items-start gap-4 mb-6">
                {phase.status === 'completed' ? (
                  <CheckCircle2 className="w-8 h-8 text-emerald-500 shrink-0 mt-1" />
                ) : phase.status === 'in-progress' ? (
                  <Activity className="w-8 h-8 text-indigo-500 shrink-0 mt-1 animate-pulse" />
                ) : (
                  <Circle className="w-8 h-8 text-slate-600 shrink-0 mt-1" />
                )}
                
                <div>
                  <h2 className="text-2xl font-bold text-white">{phase.title}</h2>
                  <p className="text-slate-500 font-mono text-sm mt-1">{phase.date}</p>
                </div>
              </div>
              
              <ul className="space-y-3 pl-12 text-slate-400">
                {phase.items.map((item, j) => (
                  <li key={j} className="flex items-center gap-3">
                    <span className="w-1.5 h-1.5 rounded-full bg-slate-600"></span>
                    {item}
                  </li>
                ))}
              </ul>
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}

function Activity(props: any) {
  return <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" {...props}><polyline points="22 12 18 12 15 21 9 3 6 12 2 12"></polyline></svg>
}
